views/default/page/layouts/widgets/add_panel.php

```diff
@@ -1,69 +1,87 @@
 <?php
 /**
  * Widget add panel
- *
- * @uses $vars['widgets']     Array of current widgets
- * @uses $vars['context']     The context for this widget layout
- * @uses $vars['exact_match'] Only use widgets that match the context
- * @uses $vars['container']   Container to optional limit widgets for. Defaults to page_owner_entity
  */
 
-$widgets = elgg_extract('widgets', $vars);
+if (!elgg_is_xhr()) {
+	echo elgg_echo('ajax:not_is_xhr');
+	return;
+}
+
 $context = elgg_extract('context', $vars);
-$exact = elgg_extract('exact_match', $vars, false);
-$container = elgg_extract('container', $vars, elgg_get_page_owner_entity());
+$exact = elgg_extract('exact_match', $vars);
+$owner_guid = (int) elgg_extract('owner_guid', $vars);
+$owner = get_entity($owner_guid);
+elgg_set_page_owner_guid($owner_guid);
```

Comment:
Make this view into a properly handled page with `elgg_ajax_gatekeeper()`. Setting page owner in views makes me icky.